app/src/androidTest/java/ch/epfl/sdp/GpsActivityTest.java

```diff
@@ -45,9 +53,26 @@
     @Rule
     public ExpectedException illegalArgument = ExpectedException.none();
 
-    private void startActivityWithBroker(LocationBroker br) {
+    private void startActivityWithBroker(LocationBroker br) throws Throwable {
```

Comment:
Can't you use a more specific exception than Throwable ?

Reply:
Nope. Throwable is the most specific object thrown by runOnUiThread()

Reply:
okay !